sesame2spiner/generate_files.cpp

```diff
@@ -273,34 +274,110 @@ void getMatBounds(int i, int matid, const SesameMetadata &metadata, const Params
+  Real ppd_factor_T = params.Get("TCoarseFactor", COARSE_FACTOR_DEFAULT_T);
+  Real ppd_factor_sie = params.Get("sieCoarseFactor", COARSE_FACTOR_DEFAULT_T);
+  Real rho_fine_diameter =
+      params.Get("rhoFineDiameterDecades", RHO_FINE_DIAMETER_DEFAULT);
+  Real TSplitPoint = params.Get("TSplitPoint", T_SPLIT_POINT_DEFAULT);
+
+  Real rho_fine_center = rhoAnchor;
+
+  // These override the rho center/diameter settings
+  Real rho_fine_min = params.Get("rhoFineMin", -1);
+  Real rho_fine_max = params.Get("rhoFineMax", -1);
+  if (rho_fine_min * rho_fine_max < 0) {
+    std::cerr << "WARNING [" << matid << "]: "
+              << "Either rhoFineMin or rhoFineMax is set while the other is still unset."
+              << " Both must be set to be sensible. Ignoring." << std::endl;
+    rho_fine_min = rho_fine_max = -1;
+  }
 
   // Forces density and temperature to be in a region where an offset
   // is not needed. This improves resolution at low densities and
   // temperatures.
 
   // Extrapolation and other resolution tricks will be explored in the
   // future.
-  if (rhoMin < STRICTLY_POS_MIN) rhoMin = STRICTLY_POS_MIN;
-  if (TMin < STRICTLY_POS_MIN) TMin = STRICTLY_POS_MIN;
+  if (rhoMin < STRICTLY_POS_MIN_RHO) rhoMin = STRICTLY_POS_MIN_RHO;
+  if (TMin < STRICTLY_POS_MIN_T) TMin = STRICTLY_POS_MIN_T;
+
+  if (piecewiseRho) {
+    if (rho_fine_min > 0) {
+      lRhoBounds = Bounds(Bounds::ThreeGrids(), rhoMin, rhoMax, rho_fine_center,
+                          rho_fine_min, rho_fine_max, ppdRho, ppd_factor_rho_lo,
+                          ppd_factor_rho_hi, true, shrinklRhoBounds);
+    } else {
+      lRhoBounds =
+          Bounds(Bounds::ThreeGrids(), rhoMin, rhoMax, rho_fine_center, rho_fine_diameter,
+                 ppdRho, ppd_factor_rho_lo, ppd_factor_rho_hi, true, shrinklRhoBounds);
+    }
+  } else {
+    lRhoBounds = Bounds(rhoMin, rhoMax, numRho, true, shrinklRhoBounds, rhoAnchor);
+  }
+  if (piecewiseT) {
+    lTBounds = Bounds(Bounds::TwoGrids(), TMin, TMax, TAnchor, TSplitPoint, ppdT,
+                      ppd_factor_T, true, shrinklTBounds);
+  } else {
+    lTBounds = Bounds(TMin, TMax, numT, true, shrinklTBounds, TAnchor);
+  }
+  if (piecewiseSie) {
+    // compute temperature as a reasonable anchor point
+    constexpr int NT = 1;
+    constexpr EOS_INTEGER nXYPairs = 2;
+    EOS_INTEGER tableHandle[NT];
+    EOS_INTEGER tableType[NT] = {EOS_Ut_DT};
+    EOS_REAL rho[2], T[2], sie[2], dx[2], dy[2];
+    {
+      eosSafeLoad(NT, matid, tableType, tableHandle, {"EOS_Ut_DT"}, Verbosity::Quiet);
+      EOS_INTEGER eospacEofRT = tableHandle[0];
+      rho[0] = rho[1] = densityToSesame(rhoAnchor);
+      T[0] = temperatureToSesame(TAnchor);
+      T[1] = temperatureToSesame(TSplitPoint);
+      eosSafeInterpolate(&eospacEofRT, nXYPairs, rho, T, sie, dx, dy, "EofRT",
+                         Verbosity::Quiet);
+      eosSafeDestroy(NT, tableHandle, Verbosity::Quiet);
+    }
```

Comment:
Does it make sense to utilize the `EOSPAC` class here directly for this lookup?

Reply:
Hmm... I'm not sure. The EOSPAC class is a lot heavier than just what I load up here. But it would be less code. Would prefer to leave this for now if you don't mind.

There might be a rewrite down the line when `sesame2spiner` becomes a lot lighter and leverages the eospac class. So, let's wait for that.

Reply:
That's fine with me then. I just hate having to deal with the EOSPAC interface